docs/design/persistent-storage.md

```diff
@@ -0,0 +1,235 @@
+
+
+```
+	
+	PersistentStore
+		TypeMeta
+		ObjectMeta
+			Name - internal name
+			Labels - labeled like Volume.PersistenceRequirements.Selector
+		Spec
+			VolumeSource -- will be backed by one of these:
+				HostPath (for a local disk implementation of storage)
+				GCEPersistentDisk { modes: ReadWriteOnce | ReadOnlyMany }
+				AWSElasticBlockStore { modes: ReadWriteOnce }
+				NFSMount { modes: ReadWriteOnce | ReadOnlyMany | ReadWriteMany }
+				*** many volume plugins
+			// volumes can be orphaned.  allow never delete.
+			RetentionPolicy StoreRetentionPolicy
+		Status
+			Phase
+			CurrentMounts	MountList
+			LastMounts		MountList
+			PodRef			ObjectReference -- the pod originally requesting persistence
+			
+
+	// Manages one type of PersistentStore
+	PersistentStoreController
+		Name
+		Spec
+			VolumeSource
+				GCEPersistentDisk { modes: ReadWriteOnce | ReadOnlyMany, resources: 5gb 30iops, .48 throughput  }
+			MaxInstances  100
+			IncrementBy   10
+		Status
+			InstanceCount
+			
+
+	//
+	//  addition to VolumeSource
+	//	
+	type MountMode byte
+	
+	const (
+		ReadWriteOnce MountMode = 1 << iota
+		ReadOnlyMany
+		ReadWriteMany
+	)
+	
+	func (vs *VolumeSource) canMount(mode MountMode) bool {
+		return vs.modes & mode != 0
+	}
+	
+```
+
+> To manually create storage, an admin would, for example, create a GCE VM, create, attach, and format 20 persistent disks.  The IDs are saved and the admin then creates 20 persistent-disk.json files and POSTs each GCE_PD to the API.  There are many ways to script adding stores to a cluster.
+
+### PersistentStoreController
+
+As a ReplicationController is to a Pod, so a PersistentStoreController is to one type of PersistentVolume.  A store controller manages all aspects of that volume's lifecycle.
+
+A controller will:
+
+* Manage a pool of storage of one type of PersistentVolume
+* Contain an interface for the dynamic provisioning of that VolumeSource (perhaps build on the volumes framework?).
+* Provisioning includes attaching the volume to a host.
+* Advise the scheduler with pod placement, keeping the volume rule logic contained to the provider. 
+* Attach disks to hosts (for cloud providers), keeping cloud credentials contained to the provider's host.
+
+Stores added manually via the API would be picked up by a running controller of that type (via a watch).  This allows manual creation and management of storage.  Some controllers will have implementations able to support dynamic creation of resources more easily than others.
+
+
+#### Pod to specify storage requirements
+
+```
+
+	Pod.Spec.Volumes.Volume 
+			Name
+			VolumeSource -- left empty by user, will be bound with available storage from the pool
+			PersistenceRequirements
```

Comment:
Similar to the scheduler looking at resource requirements, node selector, etc.

I'd call it VolumeRequirements or StorageRequirements rather than PersistenceRequirements. It's going to need to include at least capacity required.